src/client/renderer/r_array.c

```diff
@@ -180,6 +182,12 @@ void R_UploadToSubBuffer(r_buffer_t *buffer, const size_t start, const size_t si
 		return;
 	}
 
+	// Don't allow null ptrs since bufferSubData does not allow it.
+	if (!data) {
+		Com_Warn("Attempted to upload null to GPU\n");
```

Comment:
Because users (and even modders) can't generate this warning (it's a programmer error), I think `Com_Warn` is awkward here. Maybe `Com_Debug` or even `Com_Error(ERROR_DROP, ..`?